### PR TITLE
redcapError / error_handling refactor

### PR DESCRIPTION
**Pull request by spgarbet**
Major refactor of error handling. Exposed several bugs and edge cases in how errors were handled.

### CODE REVIEW

**Comment by obregos** [comment]
[ FAIL 0 | WARN 0 | SKIP 13 | PASS 2074 ]

File: R/makeApiCall.R [hunk 1]
```diff
@@ -139,7 +140,8 @@
 makeApiCall <- function(rcon, 
                         body   = list(), 
                         config = list(),
-                        url    = NULL)
+                        url    = NULL,
+                        success_status_codes = 200L)
```
Comment (couthcommander) [inline R/makeApiCall.R:144]: Does "success_status_codes" need `assert_integerish`? I'm not sure it's necessary, but it seems most arguments are checked?

**Review by couthcommander — APPROVED** [review]
This is a good change that removes complexities and redundancies. I don't see anything that stands out, and tests pass.

File: R/makeApiCall.R (hunk 1 above)
Comment (spgarbet) [reply R/makeApiCall.R:144]: Good catch.

**Comment by spgarbet** [comment]
══ Results ════════════════════════════════════
Duration: 860.2 s

── Skipped tests (9) ──────────────────────────
• Tests to be developed for 3.0.0 release (9):
  test-151-exportRecords-ArgumentValidation.R:6:5,
  test-151-exportRecords-Functionality.R:6:5,
  test-152-exportRecordsWithDags.R:6:5,
  test-153-exportRecordsWithEvents.R:6:5,
  test-155-exportRecordsWithRepeatingInstruments.R:6:5,
  test-180-exportRecordsOffline-ArgumentValidation.R:6:5,
  test-180-exportRecordsOffline-Functionality.R:6:5,
  test-190-exportReports-ArgumentValidation.R:6:5,
  test-190-exportReports-Functionality.R:6:5

[ FAIL 0 | WARN 0 | SKIP 9 | PASS 2080 ]

**Pull request merged**